.github/workflows/ci.yml

```diff
@@ -20,12 +26,22 @@ jobs:
         uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
         with:
           go-version: ${{ matrix.go }}
+      
+      - name: Install Consul (consul-version=${{ matrix.consul-version }}) for integration testing
+        shell: bash
+        run: |
+          CONSUL_VERSION="${{ matrix.consul-version }}"
+          FILENAME="consul_${CONSUL_VERSION}_linux_amd64.zip"
+          curl -sSLO "https://releases.hashicorp.com/consul/${CONSUL_VERSION}/${FILENAME}" && \
+            unzip "${FILENAME}" -d /usr/local/bin && \
+            rm "${FILENAME}"
+          consul version
 
-      - name: Install Consul, Vault and Nomad for integration testing
+      - name: Install Vault and Nomad for integration testing
```

Comment:
removed consul from this list into a separate step

Reply:
we could move this install step below and instead just use "apt-get install consul${{ matrix.ent_tag}}", where ent_tag: "", "-enterprise", so we test the latest linux packages and don't have to keep coming back to bump the version

Reply:
actually decided to do it this way because everytime there was a new consul version, the CI will start failing. 

Locking it to something specific allows us to be more intentional about when we upgrade consul in this repo rather than have to manually change the test everytime there is a new release. This also follows the same pattern as consul-ecs and a few other repos

Reply:
Do you still think this should be changed? or is this fine like this?

Reply:
But is the test fragile enough to start failing on most version upgrades? I guess I would have bigger concerns about that. We didn't previously lock in the CE consul version, nor do we currently lock in the vault and nomad versions - so I would be interested to know more about why the new tests require it.

Reply:
oh i was talking about existing test: https://github.com/search?q=repo%3Ahashicorp%2Fconsul-template%201.17.0&type=code but our chat on slack made sense so I will update this